client/src/templates/layouts/NavigationLayout.js

```diff
@@ -288,6 +297,20 @@ const NavigationLayout = (props) => {
     { stateKey: "generalFeedback", label: t("feedback.otherFeedback") },
   ];
 
+  useEffect(() => {
+    if (!isAuthenticated || !props.user) return;
+    (async function initialProfileFetch() {
```

Comment:
```suggestion
    refetchUser();
```

Instead I'd dispatch this action (which will update the global user) similar to https://github.com/FightPandemics/FightPandemics/blob/1a471543dca12ccb00a7b63367ab3ef58356e2de/client/src/pages/ProfileCompleted.js#L23